### ISSUE TITLE
How can I Verify the result from myself RGBD data.Can you provide the demo?

### ISSUE THREAD

**Issue by chenyu20230702** [ru1ven/KeypointFusion#2]
I want to infer my rgb+depth data using your model, Can you provide the demo process?

**Comment by ru1ven** [comment]
Hello! Inference on raw RGB-D data is similar to that on HO3D. Since HO3D does not release ground-truth for its evaluation set, we infer the hand root joints from a network in advance, as shown in [loader.py](https://github.com/ru1ven/KeypointFusion/blob/main/dataloader/loader.py#L1224). And, if using in-the-wild RGB-D data (not hand-only), you need to predict a bounding box to crop the hand-only images. We provide validation on HO3D as well as visualization of hand joints in [train.py](https://github.com/ru1ven/KeypointFusion/blob/main/train.py/#L402).

**Comment by chenyu19880302** [comment]
Thanks to answer my question! I have carefully read your code,but I still have a bit of confusion.
for example:
imgD = self.normalize_img(depth_crop.max(), depth_crop, center_xyz, self.cube_size)
In the function,the  center_xyz value is obtained from groundtruth. If using in-the-wild RGB-D data ,how can I get it?
another question, how can I predict the root joints ?

**Comment by ru1ven** [comment]
For validation on HO3D and NYU, we use a resnet-18 to regress the root joints in an offline manner, and use them to crop the cubes around the hand in RGB-D images, following the previous depth-based method. This is also applicable to in-the-wild scenarios.

In addition, we will update a simple demo for in-the-wild inference. Welcome to follow our code!

**Comment by git-xuefu** [comment]
> For validation on HO3D and NYU, we use a resnet-18 to regress the root joints in an offline manner, and use them to crop the cubes around the hand in RGB-D images, following the previous depth-based method. This is also applicable to in-the-wild scenarios.
> 
> In addition, we will update a simple demo for in-the-wild inference. Welcome to follow our code!

I run the demo code and find that the inference time is a little long, how can I shorten the inference time to reach the millisecond level said in the paper.

**Comment by ru1ven** [comment]
> I run the demo code and find that the inference time is a little long, how can I shorten the inference time to reach the millisecond level said in the paper.

My performance is 40-50 FPS on 4090, how is yours?

**Comment by git-xuefu** [comment]
> > I run the demo code and find that the inference time is a little long, how can I shorten the inference time to reach the millisecond level said in the paper.
> 
> My performance is 40-50 FPS on 4090, how is yours?

I just run demo_RGBD and the inference time was about 1.6 seconds in 4060

**Comment by ru1ven** [comment]
> I just run demo_RGBD and the inference time was about 1.6 seconds in 4060

It's strange. Please check the code. The inference process should be able to run in real-time on a desktop GPU

**Comment by git-xuefu** [comment]
> > I just run demo_RGBD and the inference time was about 1.6 seconds in 4060
> 
> It's strange. Please check the code. The inference process should be able to run in real-time on a desktop GPU

Thank you for your reply ,it help me a lot ,I run in 4060 laptop GPU.Maybe I need a better gpu. Is there anything else I can do to improve the inference speed , such as inference in smaller stages.

**Comment by ru1ven** [comment]
> Thank you for your reply ,it help me a lot ,I run in 4060 laptop GPU.Maybe I need a better gpu. Is there anything else I can do to improve the inference speed , such as inference in smaller stages.

Even on 4060, the performance of 1.2s is abnormal. I believe there are some unexpected errors. You can infer a depth-only baseline by discarding any RGB-D fusion module, and ensure this baseline is real-time.

**Comment by git-xuefu** [comment]
> > Thank you for your reply ,it help me a lot ,I run in 4060 laptop GPU.Maybe I need a better gpu. Is there anything else I can do to improve the inference speed , such as inference in smaller stages.
> 
> Even on 4060, the performance of 1.2s is abnormal. I believe there are some unexpected errors. You can infer a depth-only baseline by discarding any RGB-D fusion module, and ensure this baseline is real-time.

Thanks to your reply! Now,it's inference time about 0.4s in 4060 when run demo_RGBD , I found that running RGB and Depth Backbone takes most of the time about 0.2s, how can I reduce this time?

**Comment by git-xuefu** [comment]
> > Thank you for your reply ,it help me a lot ,I run in 4060 laptop GPU.Maybe I need a better gpu. Is there anything else I can do to improve the inference speed , such as inference in smaller stages.
> 
> Even on 4060, the performance of 1.2s is abnormal. I believe there are some unexpected errors. You can infer a depth-only baseline by discarding any RGB-D fusion module, and ensure this baseline is real-time.

Thanks to your reply,it help me a lot. Now, it takes about 0.4s to run demo_RGBD in depth-only baseline in 4060. I found the rgb backbone and depth backbone take about 0.2s. Do you think this is normal,how can I reduce inference time of tihs?
